### DISCUSSION TITLE
How to change nbits for IVFPQ

### DISCUSSION THREAD

**Discussion by AyushP123** [milvus-io/milvus#3909] [Q&A and General discussion] [answered]
Milvus CPU: v0.10.3 ( Built from Source ).

I want to change nbits for the IVFPQ CPU version from nbits = 8 to nbits = 4 or nbits = 2. However, I am not able to do that with Milvus. Here is the trace of my activity.

I create an index on an empty collection with the following parameters 
```
index_param = {
        "m": 64,
        "nlist": 32,
        "nbits": 4
        }
status = client.create_index(collection_name, IndexType.IVF_PQ, index_param)
```

I see the status of the index creation as 
```
Status(code=0, message='Build index successfully!')
Create Index Time 0.0040454864501953125
```
After this, I inserted 5 million vectors into the collection and saw that the row count is 0.
`(Status(code=0, message='Success'), {'partitions': [{'row_count': 0, 'segments': None, 'tag': '_default'}], 'row_count': 0})`

It seems that Milvus does not recognize nbits as a parameter, hence even though there is a message of 'Build Index Successfully" no index is actually built. Please suggest what should I do to change nbits for IVFPQ.

Suggestion: It seems that you can create an index by passing any string as a parameter and instead of throwing an error, Milvus proceeds with the index creation. It will be better to throw an error to let the user know that a wrong parameter has been passed ( You can recreate this by passing "random" with value set to 1 to IVFPQ and see that the Index is created ).

**Comment by shiyu22** [comment, marked as answer]
In fact, IVFPQ does not support modifying the `nbits` parameter, please refer to the [vector index](https://milvus.io/docs/v0.10.3/index.md) for the specific parameters of PQ.

Regarding the insertion failure, please refer to [examp.py](https://github.com/milvus-io/pymilvus/blob/master/examples/example.py) to check if there is a problem with the data format.

**Reply by sahuang** [reply]
To clarify: Until v0.11, Milvus will not support modifying `nbits` parameter during creating index.

However, we plan to add this feature in v0.12.0.
